ts/session/sending/MessageQueue.ts

```diff
@@ -143,23 +158,28 @@ export class MessageQueue implements MessageQueueInterface {
           await jobQueue.addWithId(messageId, async () =>
             MessageSender.send(message)
           );
-          void this.pendingMessageCache.remove(message);
           this.events.emit('success', message);
         } catch (e) {
           this.events.emit('fail', message, e);
+        } finally {
+          // Remove from the cache because retrying is done in the sender
+          void this.pendingMessageCache.remove(message);
```

Comment:
pendingMessageCache is saved on db, right? What happens if the message is removed from the cache, but sending fails and we quit the app?

Reply:
Ideally the `fail` event is fired before we remove from db. This event should be handled by the conversation and mark the message as failed to send. From there the user can retry